src/Microsoft.Identity.Client/ApiConfig/SystemWebViewOptions.cs

```diff
@@ -68,5 +73,85 @@ internal void LogParameters(ICoreLogger logger)
+        /// Use Microsoft Edge to navigate to the given uri. On non-windows platforms it uses 
+        /// whatever browser is the default
+        /// </summary>
+        /// <remarks>This helper is experimental and the signature may change without a major version increment.</remarks>
+        public static async Task OpenWithEdgeBrowserAsync(Uri uri)
+        {
+            if (uri == null)
+            {
+                throw new ArgumentNullException(nameof(uri));
+            }
+
+            string url = uri.AbsoluteUri;
+
+#if DESKTOP
+            url = url.Replace("&", "^&");
+            Process.Start(new ProcessStartInfo("cmd", $"/c start microsoft-edge:{url}") { CreateNoWindow = true });
+            await Task.FromResult(0).ConfigureAwait(false);
+#else
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                url = url.Replace("&", "^&");
+                Process.Start(new ProcessStartInfo("cmd", $"/c start microsoft-edge:{url}") { CreateNoWindow = true });
+            }
+            else
+            {
+                var proxy = PlatformProxyFactory.CreatePlatformProxy(new NullLogger());
+                await proxy.StartDefaultOsBrowserAsync(url).ConfigureAwait(false);
+            }
+#endif
+        }
+
+
+
+        /// <summary>
+        /// Use Microsoft Edge Chromium to navigate to the given uri. Requires the browser to be installed.
+        /// On Linux, uses the default system browser instead, as Edge is not available.
+        /// </summary>
+        /// <remarks>This helper is experimental and the signature may change without a major version increment.</remarks>
+        public static async Task OpenWithChromeEdgeBrowserAsync(Uri uri)
+        {
+            if (uri == null)
+            {
+                throw new ArgumentNullException(nameof(uri));
+            }
+
+            string url = uri.AbsoluteUri;
+
+#if DESKTOP
+            url = url.Replace("&", "^&");
+            Process.Start(new ProcessStartInfo("cmd", $"/c start msedge {url}") { CreateNoWindow = true });
+            await Task.FromResult(0).ConfigureAwait(false);
+#else
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                url = url.Replace("&", "^&");
+                Process.Start(new ProcessStartInfo("cmd", $"/c start msedge {url}") { CreateNoWindow = true });
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                var proxy = PlatformProxyFactory.CreatePlatformProxy(new NullLogger());
+                await proxy.StartDefaultOsBrowserAsync(url).ConfigureAwait(false);
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.OSX))
+            {
+                Process.Start("msedge", url);
+            }
+            else
+            {
+                throw new PlatformNotSupportedException(RuntimeInformation.OSDescription);
+            }
+#endif
```

Comment:
seems like some repeated code. can you move some of this to another method and keep on the edge/chrome specific stuff here?